plugins/outputs/http/http.go

```diff
@@ -260,6 +268,35 @@ func (h *HTTP) writeMetric(reqBody []byte) error {
 		req.SetBasicAuth(h.Username, h.Password)
 	}
 
+	// Authorization Code Grant
+	if h.CredentialsFile != "" {
+		claims := jwtGo.RegisteredClaims{}
+		_, err := jwtGo.ParseWithClaims(h.HTTPClientConfig.AccessToken, &claims, func(token *jwtGo.Token) (interface{}, error) {
+			return nil, nil
+		})
+		if err != nil {
+			return err
+		}
+
+		// Request new token if expired
+		if !claims.VerifyExpiresAt(time.Now(), true) {
+			// token is expired
+			ctx := context.Background()
+			ctx, cancel := context.WithTimeout(ctx, time.Duration(h.Timeout))
+			defer cancel()
+
+			err = h.OAuth2Config.GetAccessToken(ctx, h.URL)
+			if err != nil {
+				return err
+			}
+		}
+
+		bearerToken := "Bearer " + h.HTTPClientConfig.AccessToken
+		req.Header.Set("Authorization", bearerToken)
+		req.Header.Set("User-Agent", internal.ProductToken())
+		req.Header.Set("Accept", "application/json")
```

Comment:
`User-Agent` is already being set later and `Accept` shouldn't be hardcodes as some API's don't accept JSON.